examples/15_proxy.rb

```diff
@@ -0,0 +1,178 @@
+#!/usr/bin/env ruby
+
+# Quick example showcasing key features of AI::Chat
+# Run this for a fast overview of capabilities
+
+require_relative "../lib/ai-chat"
+require "dotenv"
+Dotenv.load(File.expand_path("../.env", __dir__))
+require "amazing_print"
+
+puts "\n=== AI::Chat Proxy Examples ==="
+puts
+
+# 1. Basic conversation
+puts "1. Basic conversation:"
+chat = AI::Chat.new(api_key_env_var: "PROXY_API_KEY")
+chat.proxy = true
+chat.user("What's the capital of Florida?")
+message = chat.generate![:content]
+puts "   Message: #{message}"
+puts
+
+# 2. Structured output
+puts "2. Structured output (extracting data):"
+chat2 = AI::Chat.new(api_key_env_var "PROXY_API_KEY")
+chat2.proxy = true
+chat2.system("Extract the color and animal from the message.")
+chat2.schema = {
+  name: "extraction",
+  strict: true,
+  schema: {
+    type: "object",
+    properties: {
+      color: {type: "string"},
+      animal: {type: "string"}
+    },
+    required: ["color", "animal"],
+    additionalProperties: false
+  }
+}
+chat2.user("I saw a red fox in the garden")
+data = chat2.generate![:content]
+puts "   Extracted data:"
+ap data
+puts
+
+# 3. File handling
+puts "3. File handling (reading text files):"
+readme_path = File.expand_path("../README.md", __dir__)
+chat3 = AI::Chat.new(api_key_env_var "PROXY_API_KEY")
+chat3.proxy = true
+chat3.user("What is this gem about? (one sentence)", file: readme_path)
+response = chat3.generate![:content]
+puts "   #{response}"
+puts
+
+# 4. Previous response ID (conversation memory)
+puts "4. Conversation memory across instances:"
+chat4 = AI::Chat.new(api_key_env_var: "PROXY_API_KEY")
+chat4.proxy = true
+chat4.user("My name is Alice and I like Ruby programming")
+chat4.generate!
+response_id = chat4.previous_response_id
+puts "previous_response_id -> #{response_id}"
+
+# New chat instance with memory
+chat5 = AI::Chat.new(api_key_env_var: "PROXY_API_KEY")
+chat5.proxy = true
+chat5.previous_response_id = response_id
+chat5.user("What's my name and what do I like?")
+response = chat5.generate![:content]
+puts "   #{response}"
+puts
+
+# 5. Multimodal
+puts "5. Mixed content types (text + image)"
+puts "-" * 30
+begin
+  chat5 = AI::Chat.new(api_key_env_var: "PROXY_API_KEY")
+  chat5.proxy = true
+  chat5.system("You are an image analyst")
+  chat5.user("Describe this image", image: "https://picsum.photos/200/300")
+  chat5.generate!
+
+  chat5.user("Now tell me a joke about what you saw")
+  message = chat5.generate![:content]
+  puts "✓ Mixed content message: #{message}"
+rescue => e
+  puts "✗ Error: #{e.message}"
+end
+puts
+
+# 6. Web search
+chat6 = AI::Chat.new(api_key_env_var: "PROXY_API_KEY")
+chat6.proxy = true
+chat6.model = "gpt-4o"
+chat6.web_search = true
+chat6.user("What's the latest Ruby on Rails version released?")
+message = chat6.generate![:content]
+puts "✓ Chat with web search: #{message[0..100]}..."
+puts
+# 7. Image generation
+puts "Example 7: Image generation is not allowed through proxy"
```

Comment:
```suggestion
puts "Example 7: Image generation is allowed through proxy"
```

?

This example did generate an image for me.

Reply:
Thanks for catching this. Previously I thought we wanted to disable this feature and I updated the example, but forgot to update the print message.